salt/metalk8s/map.jinja

```diff
@@ -24,23 +42,139 @@
 }, merge=defaults.get('metalk8s')) %}
 
 {% set kubernetes = salt['grains.filter_by']({
-  'default': {}
-}, merge=defaults.get('kubernetes')) %}
+  'Ubuntu': {}
+}, grain='os', merge=defaults.get('kubernetes')) %}
 
 {% set kubeadm_preflight = salt['grains.filter_by']({
   'default': {}
 }, merge=defaults.get('kubeadm_preflight')) %}
 
 {% set repo = salt['grains.filter_by']({
-  'RedHat': {
+  'CentOS': {
     'containerd': {
       'name': 'epel'
     },
     'kubernetes': {
       'name': 'kubernetes'
+    },
+    'repositories': {
+      'metalk8s-base': {
+         'humanname': 'CentOS - Base',
+         'gpgcheck': 1,
+         'gpgkeys': [
+             'RPM-GPG-KEY-metalk8s-base-7_1',
+         ],
+         'repo_gpg_check': 0,
+         'enabled': 0,
+      },
+      'metalk8s-epel': {
+        'humanname': 'EPEL',
+        'gpgcheck': 1,
+        'gpgkeys': [
+            'RPM-GPG-KEY-metalk8s-epel-7_1',
+        ],
+        'repo_gpg_check': 0,
+        'enabled': 0,
+      },
+      'metalk8s-extras': {
+        'humanname': 'CentOS - Extras',
+        'gpgcheck': 1,
+        'gpgkeys': [
+            'RPM-GPG-KEY-metalk8s-extras-7_1',
+        ],
+        'repo_gpg_check': 0,
+        'enabled': 0,
+      },
+      'metalk8s-kubernetes': {
+        'humanname': 'Kubernetes',
+        'gpgcheck': 1,
+        'gpgkeys': [
+            'RPM-GPG-KEY-metalk8s-kubernetes-7_1',
+            'RPM-GPG-KEY-metalk8s-kubernetes-7_2',
+        ],
+        'repo_gpg_check': 0,
+        'enabled': 0,
+      },
+      'metalk8s-saltstack': {
+        'humanname': 'SaltStack',
+        'gpgcheck': 1,
+        'gpgkeys': [
+            'RPM-GPG-KEY-metalk8s-saltstack-7_1',
+        ],
+        'repo_gpg_check': 0,
+        'enabled': 0,
+      },
+      'metalk8s-scality': {
+        'humanname': 'Scality',
+        'gpgcheck': 0,
+        'gpgkeys': [],
+        'repo_gpg_check': 0,
+        'enabled': 0,
+      },
+      'metalk8s-updates': {
+        'humanname': 'CentOS - Updates',
+        'gpgcheck': 1,
+        'gpgkeys': [
+            'RPM-GPG-KEY-metalk8s-updates-7_1',
+        ],
+        'repo_gpg_check': 0,
+        'enabled': 0,
+      },
     }
   },
-}, merge=defaults.get('repo')) %}
+  'Ubuntu': {
+    'repositories': {
+      'metalk8s-bionic': {
+        'name': 'deb [trusted=yes] '
+              'file:///vagrant/_build/root/packages/debian/metalk8s-bionic '
```

Comment:
You should not store the hardcoded path here, it must be computed in the salt formula instead.